### Teste t pra media populacional

In [13]:
from scipy.stats import norm, shapiro, ttest_1samp
import numpy as np

#### Exercício 1

mi = pH medio do solo = 7 <br><br>

H0: mi = 7 <br>
H1: mi < 7 (pH ácido)<br><br>

Ou rejeitamos H0 (temos evidências dos dados para rejeitar a hipótese nula), ou não temos evidência e supomos que ela é verdadeira. <br><br>



In [3]:
phs = [5.8, 6.3, 6.9, 6.2, 5.5]

Nenhum valor chega a 7.0 então podemos supor que afirmação do pesquisador seja verdadeira. Porém não podemos afirmar apenas olhando para os dados.

#### Verificar a normalidade dos dados

Para o teste estamos supondo que os dados seguem uma distribuição normal então faz sentido testarmos isso. Se o teste não corroborar a normalidade não podemos fazer um teste t.<br><br>

Teste de Shapiro-Wilk
- H0: os dados seguem distribuição normal
- H1: os dados não seguem dist. normal<br><br>

p-valor < 0.05 então rejeita-se H0 <br>
p-valor > 0.05 então falha-se ao rejeitar H0

In [5]:
shapiro_test = shapiro(phs)

In [11]:
shapiro_test.statistic

np.float64(0.9761208610413159)

In [12]:
shapiro_test.pvalue

np.float64(0.9128796682967908)

Com p-valor em 0.9129 não rejeita-se H0 ao nível de significância de 5% e concluímos que os dados seguem distribuição normal. Estamos aptos para realizar o teste.

In [31]:
mu = 7
resultado = ttest_1samp(phs, popmean=mu)

t_stat = resultado.statistic
p_val = resultado.pvalue / 2 if t_stat < 0 else 1 - resultado.pvalue

print(f'Estatística t: {t_stat}')
print(f'p-valor unilateral (média < {mu}): {p_val}')

Estatística t: -3.614849624699108
p-valor unilateral (média < 7): 0.011229818874029861


Conclusão: Como p-valor < 0.05, então rejeita-se H0 ao nível de significância de 5% e conclui-se que o pH médio do solo é ácido.

#### Exercício 2

In [37]:
falha = "19.8 18.5 17.6 16.7 15.8 15.4 14.1 13.6 11.9 11.4 11.4 8.8 7.5 15.4 15.4 19.5 14.9 12.7 11.9 11.4 10.1 7.9".split()
falha = [float(x) for x in falha]

In [25]:
len(falha)

22

H0: mu = 10<br>
H1: mu > 10

O enunciado diz que segue uma distribuição normal então não vamos nos valer do teste shapiro-wilk.

In [36]:
mu = 10
resultado = ttest_1samp(falha, popmean=mu)

t_stat = resultado.statistic
p_val = resultado.pvalue / 2 if t_stat > 0 else 1 - resultado.pvalue

print(f'Estatística t: {t_stat}')
print(f'p-valor unilateral (média > {mu}): {p_val}')

Estatística t: 4.90168210121239
p-valor unilateral (média > 10): 3.7812725934482336e-05


Concusão: como p-valor < 0.05 (0.000037812), então rejeita-se H0 ao nível de significância de 5% e conclui-se que a carga média na falha é maior que 10MPa.

Lembrando que sempre a hipótese e feita sempre antes de fazer o experimento e a coleta dos dados.